inspector/contracts/InspectorConsumer.sol

```diff
@@ -0,0 +1,127 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.16;
+
+import {InspectorConsumerBase} from "./InspectorConsumerBase.sol";
+import {IAggregatorRouter} from "@bisonai/orakl-contracts/src/v0.1/interfaces/IAggregatorRouter.sol";
+import {Orakl} from "@bisonai/orakl-contracts/src/v0.1/libraries/Orakl.sol";
+import {IVRFCoordinator} from "@bisonai/orakl-contracts/src/v0.1/interfaces/IVRFCoordinator.sol";
+
+contract InspectorConsumer is InspectorConsumerBase{
+    using Orakl for Orakl.Request;
+
+    uint256 public sRandomWord;
+    uint128 public sResponse;
+    address private sOwner;
+
+    IVRFCoordinator vrfCoordinator;
+    IAggregatorRouter internal router;
+
+    error OnlyOwner(address notOwner);
+
+    modifier onlyOwner() {
+        if (msg.sender != sOwner) {
+            revert OnlyOwner(msg.sender);
+        }
+        _;
+    }
+
+    constructor(address aggregatorRouter, address _rrCoordinator, address _vrfCoordinator)InspectorConsumerBase(_vrfCoordinator, _rrCoordinator){
+        sOwner = msg.sender;
+        router = IAggregatorRouter(aggregatorRouter);
+        vrfCoordinator = IVRFCoordinator(_vrfCoordinator);
+    }
+
+    // Receive remaining payment from requestDataPayment
+    receive() external payable {}
+
+    function requestDataFeed(string calldata pair) public view returns (uint80 roundId, int256 answer){
```

Comment:
How about we just read from each AggregatorProxy to save some gas?

Reply:
it's a view function so it shouldn't cost any gas, or am I missing your point?

Reply:
aha! gotcha, My bad!

Reply:
Also, I dont think this function matters that much, we can still read from a datafeed if we failed to sumit rounds.

Reply:
I see, guess it could be removed